for-developers/walkthrough/dev-setup/making-your-first-changes.md

```diff
@@ -47,4 +47,214 @@ Now you should see some custom content in the post that you added to the feed. Y
+The first set of changes consists of adding constants to identify the new transaction type you are creating.
+
+In `network.go`, perform the following changes:
+* Add a new `TxnType` constant named `TxnTypeReact` and update the value of `NEXT_ID` to `27`.
+```go
+TxnTypeDAOCoinTransfer              TxnType = 25
+TxnTypeReact                        TxnType = 26
+
+// NEXT_ID = 27
+)
+```
+* Add a new `TxnString` constant named `TxnStringReact`.
+```go
+TxnStringDAOCoinTransfer              TxnString = "DAO_COIN_TRANSFER"
+TxnStringReact                        TxnString = "REACT"
+TxnStringUndefined                    TxnString = "TXN_UNDEFINED"
+)
+```
+
+Now, update functions, methods, and constants that use these constant values.
+
+* Append the `TxnTypeReact` value to the `AllTxnTypes` `[]TxnType` slice.
+```go
+AllTxnTypes = []TxnType{
+TxnTypeUnset, ...,
+TxnTypeDAOCoin, TxnTypeDAOCoinTransfer, TxnTypeReact,
+}
+```
+* Append the `TxnStringReact` value to the `AllTxnString` `[]TxnString` slice.
+```go
+AllTxnString = []TxnString{
+TxnStringUnset, ...,
+TxnStringDAOCoin, TxnStringDAOCoinTransfer, TxnStringReact,
+}
+```
+* In `GetTxnString`, add a case for the `TxnTypeReact`.
+```go
+case TxnTypeDAOCoinTransfer:
+    return TxnStringDAOCoinTransfer
+case TxnTypeReact:
+    return TxnStringReact
+default:
+    return TxnStringUndefined
+```
+* (_Optional_) In `GetTxnTypeFromString`, add a case for the `TxnStringReact`.
+```go
+case TxnStringDAOCoinTransfer:
+    return TxnTypeDAOCoinTransfer
+case TxnStringReact:
+    return TxnTypeReact
+default:
+    // TxnTypeUnset means we couldn't find a matching txn type
+    return TxnTypeUnset
+}
+```
+
+Next, create a new transaction metadata type that records emoji reactions.
+In this example, you can assume that a single transaction contains only one emoji reaction.
+Multiple reaction to a single post would correspond to multiple transactions.
+
+* At the end of the file, insert the following code block.
+The new struct has a rune field to store the Unicode of the emoji reaction.
+The `ToBytes` method validates the post hash and
+then builds a byte slice containing both the validated post hash and the normalized NFC Unicode emoji reaction.
+* Why use a normalized NFC Unicode?
+It is possible for a single character to be encoded with different code point sequences.
+By normalizing the unicode, we ensure that a character will have a unique code point sequence.
+
+```go
+// ==================================================================
+// ReactMetadata
+//
+// A reaction is an interaction where a user on the platform reacts to a post.
+// ==================================================================
+
+type ReactMetadata struct {
+    // The user reacting is assumed to be the originator of the
+    // top-level transaction.
+    
+    // The post hash to react to.
+    PostHash *BlockHash
+    
+    EmojiReaction rune
```

Comment:
I think we also want a boolean indicating whether or not we are "removing" the reaction - similar to the `IsUnlike` in the `LikeMetadata` struct